cmd/podman/networks/list.go

```diff
@@ -88,19 +90,36 @@ func networkList(cmd *cobra.Command, args []string) error {
 		row += "\n"
 	}
 
-	format := "{{range . }}" + row + "{{end}}"
-	if !cmd.Flag("format").Changed {
-		format = headers + format
+	var headers, format string
+
+	if cmd.Flags().Changed("format") {
+		headers = ""
+		format = report.NormalizeFormat(networkListOptions.Format)
+	} else {
+		headers, format = createListOut()
 	}
+	format = headers + "{{range . }}" + row + "{{end}}"
 	tmpl, err := template.New("listNetworks").Parse(format)
 	if err != nil {
 		return err
 	}
+
 	w := tabwriter.NewWriter(os.Stdout, 8, 2, 2, ' ', 0)
+	defer w.Flush()
+
 	if err := tmpl.Execute(w, nlprs); err != nil {
 		return err
 	}
-	return w.Flush()
+	return nil
+}
+
+func createListOut() (string, string) {
```

Comment:
How about?

```
func createListOut() (string, string) {
	row := "{{.Name}}\\t{{.Version}}\\t{{.Plugins}}\\n"
	return report.NormalizeFormat(defaultHeaders), report.NormalizeFormat(row)
}
```

Reply:
You could pull the headers into here as well.